### PR TITLE
Docs: environment variables

### PR DESCRIPTION
**Pull request by stsewd**
I have moved this content from the guide since this is a feature.

Closes https://github.com/readthedocs/readthedocs.org/issues/8159

I'll create a redirect when this is approved

preview at https://docs--8390.org.readthedocs.build/en/8390/environment-variables.html

### CODE REVIEW

**Review by humitos — APPROVED** [review]
LGTM!

File: docs/environment-variables.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
+Environment Variables
+=====================
+
+It may happen that your documentation depends on an authenticated service to be built properly.
```
Comment (astrojuanlu) [inline docs/environment-variables.rst:4]: This introductory paragraph is weird. On one hand, it encourages users to use this functionality for secrets, while at the end of the guide it's shown how to reach such secrets. Wouldn't this allow an attacker to send a malicious pull request and read such secrets?

On the other hand, even if I misunderstood something and this is not a concern, I would put the "Read the Docs provides a way to define..." first, then this as an example.

Comment (humitos) [reply docs/environment-variables.rst:4]: I just read this and I agree that we should warn the user about this on the point 

> #. Check the :guilabel:`Public` option if you want to expose this environment variable
>   to :doc:`builds from pull requests </pull-requests>`.

and modify the first paragraph as well.

Comment (stsewd) [reply docs/environment-variables.rst]: Changed :+1:

**Review by astrojuanlu — APPROVED** [review]
Let's get this merged

**Pull request merged**